app/backends/workflow-orchestration.ts

```diff
@@ -755,10 +760,118 @@ export async function handleWorkItemUpdate(
+ * @param queueType - Type of the queue to read from
+ */
+export async function batchProcessQueue(queueType: WorkItemUpdateQueueType): Promise<void> {
+  const queue = getQueue(queueType);
+  // use a smaller batch size for the large item update queue otherwise use the SQS max batch size
+  // of 10
+  const largeItemQueueBatchSize = Math.min(env.largeWorkItemUpdateQueueMaxBatchSize, 10);
+  const otherQueueBatchSize = 10; // the SQS max batch size
+  const queueBatchSize = queueType === WorkItemUpdateQueueType.LARGE_ITEM_UPDATE
+    ? largeItemQueueBatchSize : otherQueueBatchSize;
+  const messages = await queue.getMessages(queueBatchSize);
+  if (messages.length < 1) {
+    return;
+  }
+  defaultLogger.debug(`Processing ${messages.length} work item updates from queue`);
+
+  if (queueType === WorkItemUpdateQueueType.LARGE_ITEM_UPDATE) {
+    // process each message individually
+    for (const msg of messages) {
+      try {
+        const updateItem: WorkItemUpdateQueueItem = JSON.parse(msg.body);
+        const { update, operation } = updateItem;
+        defaultLogger.debug(`Processing work item update from queue for work item ${update.workItemID} and status ${update.status}`);
+        await handleWorkItemUpdate(update, operation, defaultLogger);
+      } catch (e) {
+        defaultLogger.error(`Error processing work item update from queue: ${e}`);
+      }
+      try {
+        // delete the message from the queue even if there was an error updating the work-item
+        // so that we don't keep processing the same message over and over
+        await queue.deleteMessage(msg.receipt);
+      } catch (e) {
+        defaultLogger.error(`Error deleting work item update from queue: ${e}`);
+      }
+    }
+  } else {
+    // potentially process all the messages at once. this actually calls `handleBatchWorkItemUpdates`,
+    // which processes each job's updates individually right now. this just leaves the possibility
+    // open for that function to be updated to process all the updates at once in a more efficient
+    // manner. It also allows us to delete all the messages from the queue at once, which is more
+    // efficient than deleting them one at a time.
+    const updates: WorkItemUpdateQueueItem[] = messages.map((msg) => JSON.parse(msg.body));
+    try {
+      await handleBatchWorkItemUpdates(updates, defaultLogger);
+    } catch (e) {
+      defaultLogger.error(`Error processing work item updates from queue: ${e}`);
+    }
+    // delete all the messages from the queue at once (slightly more efficient)
+    try {
+      await queue.deleteMessages(messages.map((msg) => msg.receipt));
+    } catch (e) {
+      defaultLogger.error(`Error deleting work item updates from queue: ${e}`);
+    }
+  }
+}
+
 /**
  * Update a work item from a service response. This function stores the update without further
  * processing and then responds quickly. Processing the update is handled asynchronously
```

Comment:
Update the comment to say the work item update is put on a queue.

Reply:
Done.